### PR TITLE
Refactor nodesyncer

### PR DESCRIPTION
**Pull request by daxmc99**
Refactor and testing commits from https://github.com/rancher/rancher/pull/20375

### CODE REVIEW

File: pkg/node/node_test.go [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+						Port:            "",
+						InternalAddress: "",
+					},
+				},
+			},
+			want: true,
+		},
+	}
+	for _, tt := range tests {
+		if got := IsNodeForNode(tt.node, tt.machine); got != tt.want {
```
Comment (dramich) [inline pkg/node/node_test.go]: Use assert libraries

File: pkg/node/node.go [hunk 2]
```diff
@@ -13,19 +13,20 @@ const (
...
+	// to handle the case when machine was provisioned first
+	if machine.Status.NodeConfig != nil {
+		if machine.Status.NodeConfig.HostnameOverride != "" {
+			return machine.Status.NodeConfig.HostnameOverride
 		}
 	}
 	return ""
 }
 
+// IsNodeForNode returns true iff node names == or node addresses are equal
```
Comment (dramich) [inline pkg/node/node.go]: `// IsNodeForNode returns true if node names or addresses are equal`

File: pkg/controllers/user/nodesyncer/nodessyncer.go [hunk 3]
```diff
@@ -587,9 +580,9 @@ func aggregateRequestAndLimitsForNode(pods []*corev1.Pod) (map[corev1.ResourceNa
 		//podName -> req/limit -> data
 		for _, pod := range pods {
 			podsData[pod.Name] = make(map[string]map[corev1.ResourceName]resource.Quantity)
-			requests, limits := getPodData(pod)
-			podsData[pod.Name]["requests"] = requests
-			podsData[pod.Name]["limits"] = limits
+			request, limit := getPodData(pod)
```
Comment (dramich) [inline pkg/controllers/user/nodesyncer/nodessyncer.go]: I don't agree with this change

File: pkg/controllers/user/nodesyncer/nodessyncer.go [hunk 4]
```diff
@@ -490,7 +482,8 @@ func statusEqualTest(proposed, existing corev1.NodeStatus) bool {
 	return true
 }
 
-func (m *NodesSyncer) convertNodeToNode(node *corev1.Node, existing *v3.Node, pods map[string][]*corev1.Pod) (*v3.Node, error) {
+// convertNodeToMachine takes a core.v1 Node, a possibly existing machine, and the pods running on the core.v1 Node
```
Comment (dramich) [inline pkg/controllers/user/nodesyncer/nodessyncer.go]: Comment should give some context on the function if needed. The sig already tells you what goes into it

File: pkg/node/node_test.go (hunk 1 above)
Comment (daxmc99) [reply pkg/node/node_test.go]: Using testify.Assert now

File: pkg/controllers/user/nodesyncer/nodessyncer.go (hunk 3 above)
Comment (daxmc99) [reply pkg/controllers/user/nodesyncer/nodessyncer.go]: Fixed shadowing but it does make the code less clear than before.

**Comment by dramich** [comment]
lgtm

**Comment by cjellick** [comment]
@daxmc99 please reference the issue that caused you to make this change (for fixing the syncing), i know it wont go to test yet bc you have the "real" fix to make, but i want both prs to have a link to the issue

File: pkg/controllers/user/nodesyncer/nodessyncer.go [hunk 5]
```diff
@@ -298,8 +290,8 @@ func (m *NodesSyncer) removeNode(machine *v3.Node) error {
 	return nil
 }
 
-func (m *NodesSyncer) updateNode(existing *v3.Node, node *corev1.Node, pods map[string][]*corev1.Pod) error {
-	toUpdate, err := m.convertNodeToNode(node, existing, pods)
+func (m *nodesSyncer) updateNode(existing *v3.Node, node *corev1.Node, pods map[string][]*corev1.Pod) error {
+	toUpdate, err := m.convertNodeToMachine(node, existing, pods)
```
Comment (cjellick) [inline pkg/controllers/user/nodesyncer/nodessyncer.go]: 🙏

**Comment by cjellick** [comment]
lgtm

**Comment by daxmc99** [comment]
Part of addressing https://github.com/rancher/rancher/issues/14184

**Pull request merged**